services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -428,6 +437,76 @@ async def pull_input_ports(
     return transferred_bytes
 
 
+@containers_router.post(
+    "/containers/directory-watcher:disable",
+    summary="Disable directory-watcher event propagation",
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def disable_directory_watcher(
+    app: FastAPI = Depends(get_application),
+) -> Response:
+    directory_watcher.disable_directory_watcher(app)
+    # SEE https://github.com/tiangolo/fastapi/issues/2253
+    return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@containers_router.post(
+    "/containers/directory-watcher:enable",
+    summary="Enable directory-watcher event propagation",
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def enable_directory_watcher(app: FastAPI = Depends(get_application)) -> Response:
+    directory_watcher.enable_directory_watcher(app)
+    # SEE https://github.com/tiangolo/fastapi/issues/2253
+    return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@containers_router.post(
+    "/containers/ports/outputs:create-dirs",
+    summary=(
+        "Creates the output directories declared by the labels. "
+        "Since the dynamic-sidecar cannot retrieve the labels "
+        "from the image, these must be provided by the "
+        "director-v2. This API is called after the outputs directory was pulled"
+    ),
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def create_output_dirs(request_mode: CreateDirsRequestItem) -> Response:
+    mounted_volumes: MountedVolumes = get_mounted_volumes()
+    outputs_path = mounted_volumes.disk_outputs_path
+    for port_key, service_output in request_mode.outputs_labels.items():
+        if is_file_type(service_output.property_type):
+            dir_to_create = outputs_path / port_key
+            dir_to_create.mkdir(parents=True, exist_ok=True)
+
+    # SEE https://github.com/tiangolo/fastapi/issues/2253
+    return Response(status_code=status.HTTP_204_NO_CONTENT)
```

Comment:
No need for this if the ``post`` decorator would define ``response_model`` as
```python
status_code=status.HTTP_204_NO_CONTENT, response_class=Response
```

see https://github.com/pcrespov/sandbox-python/blob/fastapi-stuff/fastapi-lib/no_content_response.py

Reply:
cool, thanks for this.